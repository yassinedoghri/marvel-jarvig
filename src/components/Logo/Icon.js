import styled from "styled-components";

const Icon = styled.img`
  height: 2.5em;
`;

export default Icon;
