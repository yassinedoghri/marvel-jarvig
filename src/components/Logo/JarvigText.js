import styled from "styled-components";

export const JarvigText = styled.h1`
    color: ${props => props.theme.colors.light.secondary};
    font-family: Bangers;
    margin: 0;
    text-align: center;
    letter-spacing: .4em;
    font-size: ${props => props.sm ? '1rem' : '2rem'};
    margin-left: ${props => props.spaceLeft ? '.5em' : '0'};
`;

export default JarvigText;