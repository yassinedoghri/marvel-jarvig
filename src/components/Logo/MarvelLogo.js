import styled from "styled-components";

const MarvelLogo = styled.img`
  height: ${props => (props.sm ? "2em" : "4.5em")};
`;

export default MarvelLogo;
