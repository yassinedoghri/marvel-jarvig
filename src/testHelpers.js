import React from "react";
import "jest-styled-components";
import renderer from "react-test-renderer";
import theme from "theme";
import { ThemeProvider } from "styled-components";

export default function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}
