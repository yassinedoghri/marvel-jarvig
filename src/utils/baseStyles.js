import { normalize } from "polished";
import { injectGlobal } from "styled-components";

export default () => injectGlobal`
    ${normalize()}

    @import url("https://fonts.googleapis.com/css?family=Bangers");
    @import url("https://fonts.googleapis.com/css?family=Roboto:400,900");
    @import url("https://fonts.googleapis.com/css?family=Press+Start+2P");
  
    * {
      font-family: "Roboto", sans-serif;
      box-sizing: border-box;
    }
    
    *:focus {
      outline: none;
    }
    
    a {
      text-decoration: none;
    }
    
    ul li, ol li {
        list-style: none;
    }
    
    html, body, #root {
      height: 100%;
    }
    
    body {
      padding: 0;
      background-color: white;
    }
`;
