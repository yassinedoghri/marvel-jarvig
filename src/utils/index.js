export { default as style } from "utils/style";
export { default as baseStyles } from "utils/baseStyles";
export { default as helpers } from "utils/helpers";
export { default as testHelpers } from "utils/testHelpers";
export { default as registerServiceWorker } from "utils/registerServiceWorker";
export { default as media } from "utils/media";
