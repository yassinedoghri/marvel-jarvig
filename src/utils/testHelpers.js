import { createMemoryHistory } from "history";
import React from "react";
import { Provider } from "react-redux";
import {
  ConnectedRouter,
  routerMiddleware,
  routerReducer
} from "react-router-redux";
import renderer from "react-test-renderer";
import ShallowRenderer from "react-test-renderer/shallow";
import rootReducer from "reducers";
import { applyMiddleware, combineReducers, createStore } from "redux";
import { ThemeProvider } from "styled-components";
import theme from "theme";

const history = createMemoryHistory("/");
const routeMiddleware = routerMiddleware(history);

const store = createStore(
  combineReducers({
    ...rootReducer,
    router: routerReducer
  }),
  applyMiddleware(routeMiddleware)
);

export function renderWithTheme(component) {
  return renderer.create(
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  );
}

export function renderInContext(component) {
  return renderer.create(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}

export function ShallowRenderInContext(component) {
  const renderer = new ShallowRenderer();
  return renderer.render(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <ThemeProvider theme={theme}>{component}</ThemeProvider>
      </ConnectedRouter>
    </Provider>
  );
}
