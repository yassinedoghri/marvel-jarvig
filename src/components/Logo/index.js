import styled from 'styled-components';
import {Link} from 'react-router-dom';
import {media} from '../../media';

import MarvelLogo from './MarvelLogo';
import JarvigText from './JarvigText';

const Logo = styled(Link)`
    margin: ${props => props.center ? '0 auto' : 'inherit'};
    display: ${props => props.inline ? 'flex' : 'block'};
    align-items: center;
    visibility: ${props => props.hidden ? 'hidden' : 'visible'};
    text-decoration: none;
    ${media.tablet`margin: ${props => props.center ? '0 auto' : 'inherit'};`}
`;

Logo.MarvelLogo = MarvelLogo;
Logo.JarvigText = JarvigText;

export default Logo;
