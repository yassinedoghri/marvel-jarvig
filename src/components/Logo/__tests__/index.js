import Logo from "components/Logo";
import React from "react";
import { MemoryRouter } from "react-router-dom";
import { renderWithTheme } from "utils/testHelpers";

describe("Logo Block", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      <MemoryRouter>
        <Logo to="/">
          <Logo.MarvelLogo />
          <Logo.JarvigText />
          <Logo.Icon />
        </Logo>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("adds mobileicon modifier", () => {
    const tree = renderWithTheme(
      <MemoryRouter>
        <Logo to="/" mobileicon="true">
          <Logo.MarvelLogo />
          <Logo.JarvigText />
          <Logo.Icon />
        </Logo>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
