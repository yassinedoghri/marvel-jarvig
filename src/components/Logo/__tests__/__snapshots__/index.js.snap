// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Block adds mobileicon modifier 1`] = `
<a
  className="sc-bxivhb cZFXvN"
  href="/"
  mobileicon="true"
  onClick={[Function]}
>
  <img
    className="sc-htpNat ljPkey"
  />
  <h1
    className="sc-bwzfXH ljyKVz"
  />
  <img
    className="sc-bdVaJa ibrkLT"
  />
</a>
`;

exports[`Logo Block renders correctly 1`] = `
<a
  className="sc-bxivhb cEytNk"
  href="/"
  onClick={[Function]}
>
  <img
    className="sc-htpNat ljPkey"
  />
  <h1
    className="sc-bwzfXH ljyKVz"
  />
  <img
    className="sc-bdVaJa ibrkLT"
  />
</a>
`;
